src/expr/src/relation/func.rs

```diff
@@ -154,6 +154,25 @@ where
     }
 }
 
+fn sum_interval<'a, I>(datums: I) -> Datum<'a>
+where
+    I: IntoIterator<Item = Datum<'a>>,
+{
+    let mut sum = Interval::default();
+    let mut empty = true;
+    for d in datums {
+        if !d.is_null() {
+            empty = false;
+            // TODO: raise EvalError::IntervalOutOfRange(format!("{accum} + {other_accum}") instead of unwrap?
+            sum = *(&sum.checked_add(&d.unwrap_interval()).unwrap());
```

Comment:
TODO: error handling

Reply:
So error handling for `sum` is something that is actually not very straight forward. Generally we sum using a larger type to make overflows less likely, but not impossible. So for example when summing `i16`s we accumulate the result in an `i64`.

There's some discussion here: https://github.com/MaterializeInc/materialize/issues/2813. It sounds like the suggestion is to convert the interval to an epoch, accumulate the epoch as an `i128`, and then convert it back to an interval (you'd likely want to justify the interval result). That would would and give back useful results, but it probably would not match PostgreSQL because you lose the fidelity of time vs days vs months.

Another suggestion could be to invent a new internal type called `LargeInterval` that has the same fields as `Interval` but with larger width types. Then we could use that to accumulate the results.

I know that @frankmcsherry and @antiguru were recently discussing overflow when summing, they may have something useful to say here.